### ISSUE TITLE
feat: gradient boosting baseline (XGBoost) for 1-min returns

### ISSUE THREAD

**Issue by SantiBarrios2002** [SantiBarrios2002/hmm-momentum-trading#81]
## Context

Gradient boosting (XGBoost/LightGBM) is the standard strong baseline for tabular prediction tasks. Unlike the LSTM, it doesn't model sequence structure — it predicts from a fixed feature vector at each time step. This tests whether temporal modeling (HMM, LSTM) adds value over a pure feature-based approach.

### `src/ml/boosting.py`

**Model:**
- XGBoost regressor or classifier on the feature matrix from #79
- Hyperparameters: max_depth, n_estimators, learning_rate, subsample
- Simple grid search or reasonable defaults

**Training:**
- Temporal train/test split (no shuffling)
- Optional: walk-forward expanding window
- Feature importance analysis

## Functions

```python
def train_xgboost(X_train, y_train, X_val=None, y_val=None,
                  task="regression", n_estimators=500, max_depth=4,
                  learning_rate=0.01, early_stopping_rounds=50):
    """Train XGBoost model. Returns model + feature importances."""

def predict_xgboost(model, X_test):
    """Generate predictions. Returns (T,) array."""

def xgboost_to_signal(predictions, threshold=0.0):
    """Convert predictions to trading signal ∈ {-1, 0, +1}."""

def feature_importance_report(model, feature_names):
    """Return sorted feature importances as dict."""
```

## Tests (`tests/test_boosting.py`)

- [ ] Model trains and predicts correct shape on synthetic data
- [ ] Feature importances sum to ~1.0
- [ ] Signal conversion produces valid {-1, 0, +1}
- [ ] Predictions are finite
- [ ] Handles NaN features gracefully (XGBoost native support)

## Acceptance criteria

- [ ] Docstrings
- [ ] All tests pass
- [ ] Feature importance tells us which features matter most
- [ ] Compatible with backtest framework

## Tech stack note

Requires `xgboost` package (add to requirements).

## Dependencies

- #79 (ML feature engineering pipeline)
